web/js/components/geosearch/geosearch-modal.js

```diff
@@ -0,0 +1,459 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import {
+  Button, InputGroup, InputGroupAddon, UncontrolledTooltip,
+} from 'reactstrap';
+import {
+  throttle as lodashThrottle,
+  get as lodashGet,
+} from 'lodash';
+import googleTagManager from 'googleTagManager';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import SearchBox from './geosearch-input';
+import Alert from '../util/alert';
+import { getCoordinateFixedPrecision, isValidCoordinates } from './util';
+import {
+  clearSuggestions,
+  getSuggestions,
+  setPlaceMarker,
+  setSuggestion,
+  toggleReverseGeocodeActive,
+  toggleShowGeosearch,
+} from '../../modules/geosearch/actions';
+import {
+  areCoordinatesWithinExtent,
+} from '../../modules/geosearch/util';
+import {
+  processMagicKey,
+  reverseGeocode,
+} from '../../modules/geosearch/util-api';
+
+class GeosearchModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isTouchDevice: false,
+      showInputAlert: false,
+      showExtentAlert: false,
+      showReverseGeocodeAlert: false,
+      showNoSuggestionsAlert: false,
+    };
+    this.throttleGetSuggestions = lodashThrottle(this.getSuggestions, 400, { leading: true, trailing: true });
```

Comment:
I think a debounce might actually make more sense for this than a throttle.  The reason being that, when using debounce no requests will be made while the user is actively typing (depending on the configured delay and the user's typing speed).  While using throttle, requests still get made every `n` milliseconds whether or not the user has finished typing.

In my test example, I typed in "Panama City, FL" and ~10 calls were made to the server during that time but the results along the way were not of any use to me (e.g. just searching for "panam")

![Screen Shot 2020-12-04 at 8 43 06 AM](https://user-images.githubusercontent.com/1480000/101170644-c060aa00-360c-11eb-86ed-81ce76bba572.png)

If this were using a debounce, and depending on my typing speed, we would likely only see ~1-2 requests being made.  

Now, these request are all really really tiny and they are coming back at a blazing fast speed so this may only really matter if there is a cap on how many requests to this endpoint we are allowed to make.  However, I think it's good API use etiquette to limit the number of requests the application makes as much as is practical, especially if there aren't any significant benefits gained by making 10x as many.

Reply:
Oh and, since I couldn't remember exactly how throttle differed I Google'd it and found this really great article with visual examples that I think is worth sharing: https://redd.one/blog/debounce-vs-throttle